File: ios/EndlessRumination/Models/Take.swift
import Foundation

struct Take: Identifiable, Codable {
    var id: UUID = UUID()
    let lensIndex: Int
    let headline: String
    let body: String
    let wise: Bool

    enum CodingKeys: String, CodingKey {
        case lensIndex = "lens_index"
        case headline
        case body
        case wise
    }

    init(lensIndex: Int, headline: String, body: String, wise: Bool = true) {
        self.lensIndex = lensIndex
        self.headline = headline
        self.body = body
        self.wise = wise
    }

    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.lensIndex = try container.decode(Int.self, forKey: .lensIndex)
        self.headline = try container.decode(String.self, forKey: .headline)
        self.body = try container.decode(String.self, forKey: .body)
        self.wise = (try? container.decode(Bool.self, forKey: .wise)) ?? true
    }

    /// Whether this take comes from a purchased voice pack (index >= 20).
    var isPackVoice: Bool { lensIndex >= 20 }

    /// The pack name this voice belongs to, if it's a pack voice.
    var packName: String? { VoicePack.pack(forVoiceIndex: lensIndex)?.name }
}

File: android/app/src/main/kotlin/com/endlessrumination/model/Take.kt
package com.endlessrumination.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Take(
    @SerialName("lens_index") val lensIndex: Int,
    val headline: String,
    val body: String,
    val wise: Boolean = true
) {
    val isPackVoice: Boolean get() = lensIndex >= 20

    val packName: String? get() = VoicePack.packForVoiceIndex(lensIndex)?.name
}
